### PR TITLE
Move methods and properties to EA Mixin classes

### PR DESCRIPTION
**Pull request by jbrockmendel**
Not the all-at-once that @jorisvandenbossche asked for, but it's a much bigger chunk than the last few.

### CODE REVIEW

**Comment by jbrockmendel** [comment]
TODO: docstrings will need to be updated, still say FooIndex

File: pandas/core/arrays/datetimes.py [hunk 1]
```diff
@@ -143,3 +185,399 @@ def to_pydatetime(self):
...
+        2   2014-12-31
+        dtype: datetime64[ns]
+        >>> dates_series.dt.is_leap_year
+        0     True
+        1    False
+        2    False
+        dtype: bool
+        """)
+
+    def to_julian_date(self):
```
Comment (jreback) [inline pandas/core/arrays/datetimes.py]: todo update doc-string

**Review by jreback — CHANGES_REQUESTED** [review]
lgtm. minor comments for followup.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
@TomAugspurger @jorisvandenbossche if any comments.

**Comment by jreback** [comment]
thank you sir!

**Pull request merged**